pyflakes/checker.py

```diff
@@ -716,6 +716,11 @@ def handleNodeLoad(self, node):
         if not name:
             return
 
+        if name == 'print':
+            whitelisted_parents = (ast.Call, ast.Assign)
```

Comment:
by whitelisting only a limited set of parent nodes, it becomes invalid in lots of other scenarios than the original issue was about.

e.g.
```py
def a():
    return print

a = lambda: print

if print: pass
```

I have seen valid code which does the first case above, in code which how a program should output error info.  It isnt hard to imagine the other two occurring in real code;  for example, any code which uses `print = log`.

There may be others which also need to be whitelisted.

It might be better to define the set of cases where `print` is semantically ridiculous, and specifically exclude them.

Either way will require some more tests to show which usage types are valid and which are not valid.